datasetinsights/datasets/groceries_real.py

```diff
@@ -217,6 +214,39 @@ def _convert_to_bbox2d(single_bbox):
         )
         return canonical_bbox
 
+    @staticmethod
+    def _preprocess_dataset(data_path):
```

Comment:
If i understand this correctly this method will always try to unzip the "dataset" file. Furthermore this method and the one in SynDetection2D is puts the dataset in a hidden path either synthetic or groceries_real. This mean that two calls to train or evaluate will give different results. The first that unzips will point to the unzipped folder, the other will point to the data_path given.

Reply:
Also can you add some tests for the construction of the GroceriesReal and SynDetection2D with the cases where decompressing happens and for calls where DatasetNotFoundError is raised.

Reply:
We can unzip directly in data_path instead of unzip_folder.